### PR TITLE
Improve utility settings

### PR DESCRIPTION
**Pull request by thisisdano**
Removes `$output-all-utilities` setting and replaces it with `$output-these-utilities`. `$output-all-utilities` is an unusual setting — it overrides any utility-specific settings to output all utilities. `$output-these-utilities` allows users to to create a uselist of all utility modules they want to include in the project. 

Previously, to disable any utility, a developer would have to change the value of its settings object. Since all utilities are output by default, disabling the status of multiple utility modules requires adding multiple settings objects, a dreary and tedious affair that looks something like this:

```
  $display-settings: (
    output: false,
  ),
  $flex-settings: (
    output: false
  ),
  $flex-direction-settings: (
    output: false
  ),
  $flex-wrap-settings: (
    output: false
  ),
  $float-settings: (
    output: false
  ),
  $font-settings: (
    output: false,
  ),

... ad infinitum
```

With `$output-these-utilities`, the default is `default` — which means "all utilities". To include only specific utilities, devs can pass a uselist of all the desired utilities:

```
  $output-these-utilities: (
    "add-list-reset",
    "align-items",
    "align-self",
    "border",
    "border-style",
    "border-width",
    "color",
    "display",
    "font",
    "font-weight",
    "order",
    "text-decoration"
    ),
```

## Also

- A couple other minor changes outlined in the notes

### CODE REVIEW

File: packages/uswds-core/src/styles/mixins/_utility-builder.scss [hunk 1]
```diff
@@ -169,192 +170,194 @@ through all possible variants
...
-          @if meta.type-of($val-value) == "map" {
-            @if not map.get($val-value, content) {
-              $val-value: false;
-            } @else if
-              map.has-key($val-value, dependency) and not
-              map.get($val-value, dependency)
-            {
-              $val-value: false;
+    // Check to see if the utility is in the output uselist
+    @if ($output-these-utilities == "default") or list.index($output-these-utilities, $utility-name) {
```
Comment (thisisdano) [inline packages/uswds-core/src/styles/mixins/_utility-builder.scss:174]: Only cycle through utilities on the uselist

File: packages/usa-identifier/src/styles/_usa-identifier.scss [hunk 2]
```diff
@@ -142,7 +142,8 @@ $identifier-links-gap: 4 !default;
   }
 }
 
-.usa-identifier__required-link {
+.usa-identifier__required-link,
+.usa-identifier__required-link.usa-link {
```
Comment (thisisdano) [inline packages/usa-identifier/src/styles/_usa-identifier.scss:146]: Prevent `usa-link` from overriding Identifier styles, regardless of where the usa-link styles fall in the cascade

File: packages/uswds-core/src/styles/_deprecated.scss [hunk 3]
```diff
@@ -17,28 +17,5 @@
...
-@mixin typeset-title {
-  @include typeset-display;
-}
-
-// Deprecated in 2.12.0
-$theme-input-tile-background-color-selected: "primary-lighter" !default;
-$theme-input-tile-border-color: "base-lighter" !default;
-$theme-input-tile-border-color-selected: "primary-lighter" !default;
+// Deprecated in 3.0.0
+$output-all-utilities: true;
```
Comment (thisisdano) [inline packages/uswds-core/src/styles/_deprecated.scss:21]: Deprecate `$output-all-utilities` setting and remove 2.x deprecated settings

File: packages/uswds-core/src/styles/_notifications.scss [hunk 4]
```diff
@@ -13,11 +13,14 @@
...
-  + "\A --------------------------------------------------------------------"
-  + "\A No notifications";
+  "\A"
++ "\A --------------------------------------------------------------------"
++ "\A \2709  USWDS Notifications"
++ "\A --------------------------------------------------------------------"
++ "\A 3.0.0"
++ "\A We deprecated the $output-all-utilities setting."
++ "\A Control utility output with $output-all-utilities."
++ "\A See designsystem.digital.com/documentation/settings for more.";
```
Comment (thisisdano) [inline packages/uswds-core/src/styles/_notifications.scss:23]: Add a deprecation notification

File: packages/uswds-core/src/styles/_notifications.scss [hunk 5]
```diff
@@ -26,8 +29,7 @@ $uswds-notification-disable-message:
 + "\A These are notifications from the USWDS team, not necessarily a"
 + "\A problem with your code."
 + "\A"
-+ "\A Disable notifications using `$theme-show-notifications: false`"
-+ "\A in your general settings file."
++ "\A Disable notifications using `$theme-show-notifications: false`."
```
Comment (thisisdano) [inline packages/uswds-core/src/styles/_notifications.scss:32]: Remove reference to multiple settings files

File: packages/uswds-utilities/src/styles/rules/add-list-reset.scss [hunk 6]
```diff
@@ -19,7 +19,7 @@ example:
 @use "uswds-core/src/styles/settings" as *;
 
 $add-list-reset: (
-  list-reset: (
+  add-list-reset: (
```
Comment (thisisdano) [inline packages/uswds-utilities/src/styles/rules/add-list-reset.scss:22]: Use proper name for this utility object key

File: packages/uswds-utilities/src/styles/utility-fonts.scss [hunk 7]
```diff
@@ -13,10 +15,14 @@ $if-important: "";
...
-    [class*="#{ns('utility')}font-#{$face}-"] {
-      font-family: #{$stack}#{$if-important};
+// Generate font rules if the `font` utility is on the uselist 
+// and its output is set to true
+@if (($output-these-utilities == "default" or list.index($output-these-utilities, "font")) and map.get($font-settings, "output") == true) {
+  @each $face, $stack in $project-font-stacks {
+    @if $stack {
+      [class*="#{ns('utility')}font-#{$face}-"] {
+        font-family: #{$stack}#{$if-important};
+      }
```
Comment (thisisdano) [inline packages/uswds-utilities/src/styles/utility-fonts.scss:25]: Only output these font rules if the `font` utility is on the uselist

File: packages/uswds-core/src/styles/mixins/_utility-builder.scss [hunk 8]
```diff
@@ -169,192 +170,194 @@ through all possible variants
...
+      // Only do this if the the utility is meant to output
+      @if not($media-key) or
+        ($media-key and map-deep-get($utility, settings, responsive))
+      {
+        @if map-deep-get($utility, settings, output) {
```
Comment (thisisdano) [inline packages/uswds-core/src/styles/mixins/_utility-builder.scss:179]: Alllll these other changes are just adding a tab indent...

File: packages/uswds-core/src/styles/mixins/_utility-builder.scss [hunk 9]
```diff
@@ -169,192 +170,194 @@ through all possible variants
...
+                  $value,
+                  $val-value,
+                  $media-key
+                );
+              } // end the modifier loop
+            } // end the null value conditional
+          } // end the value loop
+        } // end the output conditional
+      }
+    } // end the uselist conditional
```
Comment (thisisdano) [inline packages/uswds-core/src/styles/mixins/_utility-builder.scss:360]: ....and the tab indent section finally ends

File: packages/uswds-core/src/theme/_uswds-theme-utilities.scss [hunk 10]
```diff
@@ -18,7 +18,7 @@ https://designsystem.digital.gov/utilities
 */
 
 $utilities-use-important: false;
-$output-all-utilities: true;
+$output-these-utilities: default !default;
```
Comment (thisisdano) [inline packages/uswds-core/src/theme/_uswds-theme-utilities.scss:21]: Whoops, this has a default flag!

Comment (thisisdano) [reply packages/uswds-core/src/theme/_uswds-theme-utilities.scss]: FIXED

File: packages/uswds-core/src/styles/_notifications.scss (hunk 4 above)
Comment (mejiaj) [reply packages/uswds-core/src/styles/_notifications.scss:23]: We should advise to use the new `$output-these-utilities` setting right?

```suggestion
+ "\A 3.0.0"
+ "\A We deprecated the $output-all-utilities setting."
+ "\A Control utility output with $output-these-utilities."
+ "\A See designsystem.digital.com/documentation/settings for more.";
```

**Review by mejiaj — COMMENTED** [review]
Question on notification, otherwise LGTM.

Pretty nice! I was able to successfully create a list of utility overrides and disable a utility breakpoint and only output those new utilities.

**Pull request merged**